src/main/java/org/jabref/model/database/DuplicationChecker.java

```diff
@@ -104,10 +104,13 @@ public void listen(EntriesRemovedEvent entriesRemovedEvent) {
     }
 
     @Subscribe
-    public void listen(EntryAddedEvent entryAddedEvent) {
-        Optional<String> citekey = entryAddedEvent.getBibEntry().getCiteKeyOptional();
-        if (citekey.isPresent()) {
-            addKeyToSet(citekey.get());
+    public void listen(EntriesAddedEvent entriesAddedEvent) {
+        List<BibEntry> entries = entriesAddedEvent.getBibEntries();
+        for (BibEntry entry : entries) {
```

Comment:
Does `entries.stream().map(BibEntry::getCiteKeyOptional).forEach(citekey -> addKeyToSet(citkey);` work (instead of the five lines of code?

Reply:
We need to account for empty `Optional`s, I think. So you would have to make it `entries.stream().map(BibEntry::getCiteKeyOptional).flatmap(Optional::stream).forEach(citekey -> addKeyToSet(citkey);`.